src/fifelse.c

```diff
@@ -344,3 +344,169 @@ SEXP fcaseR(SEXP na, SEXP rho, SEXP args) {
   UNPROTECT(nprotect);
   return ans;
 }
+
+SEXP psumR(SEXP na, SEXP args) {
+  if (!IS_TRUE_OR_FALSE(na)) {
+    error(_("Argument 'na.rm' must be TRUE or FALSE and length 1."));
+  }
+  const int n=length(args);
+  if (n <= 1) {
+    error(_("Please supply at least 2 arguments. (%d argument supplied)"), n);
```

Comment:
why error, instead of just returning x itself?

Reply:
I find it odd to provide 1 argument. A sum operation needs at least 2 arguments. Does it make sense to you to only provide 1 argument?

Reply:
yes, it's odd, but edge case like this can come up in dynamic scripting, where e.g. user is running

```
DT[ , do.call(psum, .SD), .SDcols = is.numeric]
```

and `DT` has a variable number of numeric columns depending on some other conditions in the script.

This way, the user can use the same code to cover the normal case when there are several normal columns, as well as the case when there only ends up being one such column.

Reply:
Ok will change that later. Tks